lib/friendly_shipping/services/ups_json/rates_options.rb

```diff
@@ -0,0 +1,122 @@
+    # @option negotiated_rates [Boolean] if truthy negotiated rates will be requested from ups. Only valid if
+    #   shipper account has negotiated rates. Default: false
+    # @option pickup_type [String] UPS pickup type. See UPS docs for more details. Default: `daily_pickup`
+    # @option pickup_date [Time] UPS pickup date/time. Default: nil
+    # @option saturday_delivery [Boolean] should we request Saturday delivery?. Default: false
+    # @option saturday_pickup [Boolean] should we request Saturday pickup?. Default: false
+    # @option shipping_method [FriendlyShipping::ShippingMethod] Request rates for a particular shipping method only?
+    #   Default is `nil`, which translates to 'All shipping methods' (The "Shop" option in UPS parlance)
+    # @option sub_version [String] The UPS API sub-version to use for requests. Default: 1707
+    # @option with_time_in_transit [Boolean] Whether to request timing information alongside the rates
+    # @option package_options_class [Class] See FriendlyShipping::ShipmentOptions
+    #
+    class UpsJson
+      class RatesOptions < FriendlyShipping::ShipmentOptions
+        PICKUP_TYPE_CODES = {
+          daily_pickup: "01",
+          customer_counter: "03",
+          one_time_pickup: "06",
+          on_call_air: "07",
+          suggested_retail_rates: "11",
+          letter_center: "19",
+          air_service_center: "20"
+        }.freeze
+
+        CUSTOMER_CLASSIFICATION_CODES = {
+          shipper_number: "00",
+          daily_rates: "01",
+          retail_rates: "04",
+          regional_rates: "05",
+          general_rates: "06",
+          standard_rates: "53"
+        }.freeze
+
+        SUB_VERSIONS = %w[v1 v1601 v1607 v1701 v1707 v2108 v2205].freeze
+
+        attr_reader :carbon_neutral,
+                    :customer_context,
+                    :destination_account,
+                    :negotiated_rates,
+                    :pickup_date,
+                    :saturday_delivery,
+                    :saturday_pickup,
+                    :shipper,
+                    :shipper_number,
+                    :shipping_method,
+                    :sub_version,
+                    :trans_id,
+                    :transaction_src,
+                    :with_time_in_transit
+
+        def initialize(
+          shipper_number:,
+          carbon_neutral: true,
+          customer_context: nil,
+          customer_classification: :daily_rates,
+          destination_account: nil,
+          negotiated_rates: false,
+          pickup_type: :daily_pickup,
+          pickup_date: nil,
+          saturday_delivery: false,
+          saturday_pickup: false,
+          shipper: nil,
+          shipping_method: nil,
+          sub_version: "v1707",
+          with_time_in_transit: false,
+          package_options_class: FriendlyShipping::Services::UpsJson::RatesPackageOptions,
```

Comment:
I think it's more readable to leave off the namespace here since it's the current namespace anyway.